lib/Containers/MerkleTree.cpp

```diff
@@ -474,16 +461,67 @@ std::unique_ptr<MerkleTree<BranchingBits, LockStripes>> MerkleTree<BranchingBits
       new MerkleTree<BranchingBits, LockStripes>(*this));
 }
 
+template <std::size_t const BranchingBits, std::size_t const LockStripes>
+std::unique_ptr<MerkleTree<BranchingBits, LockStripes>>
+MerkleTree<BranchingBits, LockStripes>::cloneWithDepth(std::size_t newDepth) {
+  std::shared_lock<std::shared_mutex> guard(_bufferLock);
+
+  if (newDepth == meta().maxDepth) {
+    return std::unique_ptr<MerkleTree<BranchingBits, LockStripes>>(
+        new MerkleTree<BranchingBits, LockStripes>(*this));
+  }
+
+  if (newDepth < meta().maxDepth) {
+    std::unique_ptr<MerkleTree<BranchingBits, LockStripes>> newTree =
+        std::make_unique<MerkleTree<BranchingBits, LockStripes>>(newDepth,
+                                                                 meta().rangeMin,
+                                                                 meta().rangeMax);
```

Comment:
```suggestion
    auto newTree =
        std::make_unique<MerkleTree<BranchingBits, LockStripes>>(newDepth,
                                                                 meta().rangeMin,
                                                                 meta().rangeMax);
```

Reply:
To be honest, I'd like to avoid the use of `auto`.